src/lib.rs

```diff
@@ -0,0 +1,704 @@
+			running,
+			config,
+			chain_access,
+			channel_manager,
+			chain_monitor,
+			peer_manager,
+			gossip_sync,
+			persister,
+			logger,
+			network_graph,
+			scorer,
+			invoice_payer,
+			_inbound_payments,
+			_outbound_payments,
+			event_queue,
+		})
+	}
+}
+
+/// Wraps all objects that need to be preserved during the run time of `LdkLite`. Will be dropped
+/// upon [`LdkLite::stop()`].
+struct LdkLiteRuntime {
+	_tokio_runtime: tokio::runtime::Runtime,
+	_background_processor: BackgroundProcessor,
+	stop_networking: Arc<AtomicBool>,
+	stop_wallet_sync: Arc<AtomicBool>,
+}
+
+/// The main interface object of the simplified API, wrapping the necessary LDK and BDK functionalities.
+///
+/// Needs to be initialized and instantiated through [`LdkLiteBuilder::build`].
+pub struct LdkLite {
+	running: RwLock<Option<LdkLiteRuntime>>,
+	config: Arc<LdkLiteConfig>,
+	chain_access: Arc<LdkLiteChainAccess<MemoryDatabase>>,
+	channel_manager: Arc<ChannelManager>,
+	chain_monitor: Arc<ChainMonitor>,
+	peer_manager: Arc<PeerManager>,
+	gossip_sync: Arc<GossipSync>,
+	persister: Arc<FilesystemPersister>,
+	logger: Arc<FilesystemLogger>,
+	scorer: Arc<Mutex<Scorer>>,
+	network_graph: Arc<NetworkGraph>,
+	invoice_payer: Arc<InvoicePayer<LdkLiteEventHandler>>,
+	_inbound_payments: Arc<PaymentInfoStorage>,
+	_outbound_payments: Arc<PaymentInfoStorage>,
+	event_queue: (EventSender, EventReceiver),
+}
+
+impl LdkLite {
+	/// Starts the necessary background tasks, such as handling events coming from user input,
+	/// LDK/BDK, and the peer-to-peer network. After this returns, the [`LdkLite`] instance can be
+	/// controlled via the provided API methods in a thread-safe manner.
+	pub fn start(&mut self) -> Result<(), Error> {
+		// Acquire a run lock and hold it until we're setup.
+		// TODO: maybe this could be done cleaner with a Condvar?
+		let mut run_lock = self.running.write().unwrap();
+		if run_lock.is_some() {
+			// We're already running.
+			return Err(Error::AlreadyRunning);
+		}
+
+		let runtime = self.setup_runtime()?;
+		*run_lock = Some(runtime);
+		Ok(())
+	}
+
+	/// Disconnects all peers, stops all running background tasks, and shuts down [`LdkLite`].
+	pub fn stop(&mut self) -> Result<(), Error> {
+		let mut run_lock = self.running.write().unwrap();
+		if run_lock.is_none() {
+			// We're not running.
+			return Err(Error::NotRunning);
+		}
+
+		let runtime = run_lock.as_ref().unwrap();
+
+		// Stop wallet sync
+		runtime.stop_wallet_sync.store(true, Ordering::Release);
+
+		// Stop networking
+		runtime.stop_networking.store(true, Ordering::Release);
+		self.peer_manager.disconnect_all_peers();
+
+		// Wait for 500ms to make sure shutdown can happen cleanly.
+		thread::sleep(Duration::from_millis(500));
+
+		// Drop the runtime, which stops the background processor and any possibly remaining tokio threads.
+		*run_lock = None;
+		Ok(())
+	}
+
+	fn setup_runtime(&self) -> Result<LdkLiteRuntime, Error> {
+		let _tokio_runtime =
```

Comment:
To make clippy shut up for the time being about the never read field. Will change it back eventually.

Reply:
Well, since we technically only ever call to the tokio runtime in this method, we only access the local variable, never the field. We still need to store it somewhere though 🤷‍♂️